cmd/clusterctl/client/cluster/workload_cluster.go

```diff
@@ -18,15 +18,17 @@ package cluster
 
 import (
 	"github.com/pkg/errors"
+	"k8s.io/klog/v2"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	utilkubeconfig "sigs.k8s.io/cluster-api/util/kubeconfig"
+	utilsecret "sigs.k8s.io/cluster-api/util/secret"
 )
 
 // WorkloadCluster has methods for fetching kubeconfig of workload cluster from management cluster.
 type WorkloadCluster interface {
 	// GetKubeconfig returns the kubeconfig of the workload cluster.
-	GetKubeconfig(workloadClusterName string, namespace string) (string, error)
+	GetKubeconfig(workloadClusterName string, namespace string, userKubeconfig bool) (string, error)
```

Comment:
Changing the function signature here will cause a breaking API code change.

Reply:
@valaparthvi could you introduce a new function? We can then refactor the existing one to call the new one.

Reply:
What about having a `GetUserKubeconfig` function, this will make things explicit while we can preserve the current API